rules/0132.go

```diff
@@ -0,0 +1,201 @@
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0132", "request_message", "unknown_fields"),
+			Description:  "check that there are no unknown fields",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/132#request-message",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about List- methods for the purpose of this rule;
+				// ignore everything else.
+				if !isListMethod(m) {
+					return
+				}
+
+				// Rule check: Establish that there are no unexpected fields.
+				allowedFields := map[string]struct{}{
+					"parent":       {}, // AIP-131
+					"page_size":    {}, // AIP-158
+					"page_token":   {}, // AIP-158
+					"filter":       {}, // AIP-132
+					"order_by":     {}, // AIP-132
+					"group_by":     {}, // Nowhere yet, but permitted
+					"show_deleted": {}, // AIP-135
+					"read_mask":    {}, // AIP-157
+					"view":         {}, // AIP-157
+				}
+				fields := m.Input().Fields()
+				for i := 0; i < fields.Len(); i++ {
+					field := fields.Get(i)
+					if _, ok := allowedFields[string(field.Name())]; !ok {
+						problems = append(problems, lint.Problem{
+							Message:    "List RPCs should only contain fields explicitly described in AIPs.",
+							Descriptor: field,
+						})
+					}
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// Get messages should use the resource as the respose message
+func checkListResponseMessageName() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0132", "response_message", "name"),
+			Description:  "check that List RPCs have appropriate response messages",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/132#guidance",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about List- methods for the purpose of this rule;
+				// ignore everything else.
+				if !isListMethod(m) {
+					return
+				}
+
+				// Rule check: Establish that for methods such as `ListFoos`, the response
+				// message is named `ListFoosResponse`.
+				methodName := string(m.Name())
+				responseMessageName := string(m.Output().Name())
+				if responseMessageName != methodName+"Response" {
+					problems = append(problems, lint.Problem{
+						Message: fmt.Sprintf(
+							"List RPCs should have a response message named after the RPC, such as %q.",
+							methodName+"Response",
+						),
+						Suggestion: methodName + "Response",
+						Descriptor: m,
+					})
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// Return true if this is a List method, false otherwise.
+func isListMethod(m p.MethodDescriptor) bool {
+	return strings.HasPrefix(string(m.Name()), "List")
```

Comment:
Borderline nitpick: I imagine the corner cases are small, but this might be safer with a regex like `List(?:[A-Z]|$)` in other words, you may hit false positives with words that begin with `List`. `Listen`, `Listener`, etc. come to mind.

Reply:
I buy that.

Reply:
Done.